client/scripts/legic.lua

```diff
@@ -1974,14 +1948,17 @@ function check4LegicCash(data)
   if(#data==32) then
     local stamp_len=(#data-25)
     local stamp=""
+    local debugCash = false
     for i=0, stamp_len-1 do
       stamp=stamp..data[i].." "
     end
     if (data[7]=="01") then
-      if (("%04x"):format(utils.Crc16(dumpTable(data, "", 0, 12))) == data[13]..data[14]) then
-        if (("%04x"):format(utils.Crc16(dumpTable(data, "", 15, 20))) == data[21]..data[22]) then
-          if (("%04x"):format(utils.Crc16(dumpTable(data, "", 23, 29))) == data[30]..data[31]) then
-            io.write(accyan.."Legic-Cash Segment detected "..acoff)
+      if (debugCash or ("%04x"):format(utils.Crc16(dumpTable(data, "", 0, 12))) ~= data[13]..data[14]) then
+        --if(debugCash) then  io.write("[1]"..dumpTable(data, "", 0, 12).." ==> "..("%04x"):format(utils.Crc16(dumpTable(data, "", 0, 12))) .."!=".. data[13]..data[14].."\n") end
+        if (debugCash  or ("%04x"):format(utils.Crc16(dumpTable(data, "", 15, 20))) ~= data[21]..data[22]) then
+          --if(debugCash) then io.write("[2]"..("%04x"):format(utils.Crc16(dumpTable(data, "", 15, 20))) .."~=".. data[21]..data[22].."\n") end
+          if (debugCash  or ("%04x"):format(utils.Crc16(dumpTable(data, "", 23, 29))) ~= data[30]..data[31]) then
+            -- io.write(accyan.."Legic-Cash Segment detected "..acoff)
             return true
```

Comment:
and here you changed from  == to  ~=   (which means from  EQUAL to NOT EQUAL). ..  
If a crc-check should work,  the crc values should be as before. EQUAL.   
So whats your purpose here? (so that I may understand your thoughs)